modin/config/envvars.py

```diff
@@ -67,6 +74,81 @@ def get_help(cls) -> str:
         return help
 
 
+class EnvWithSibilings(
+    EnvironmentVariable,
+    # we have to pass anything here in order to derive from 'EnvironmentVariable',
+    # this doesn't force child classes to have 'str' type, they actually can be any type
+    type=str,
```

Comment:
I don't quite get why we have to pass anything here? We already specify `type=str` in the declaration of EnvironmentVariable.

Reply:
idk, it just doesn't work without it:
```python
from modin.config import EnvironmentVariable

class Child1(EnvironmentVariable, type=str): # creates successfuly
    pass

class Child2(EnvironmentVariable):
    pass
# Traceback (most recent call last):
#   File "modin/t4.py", line 6, in <module>
#     class Child2(EnvironmentVariable):
# TypeError: __init_subclass__() missing 1 required positional argument: 'type'
```

Reply:
I see. We could just make `type` parameter of `__init_subclass__` as a keyword argument and set `str` as a default value for it.

Reply:
Actually, what I said above is not best solution. We could add some `AnyPossible` type, which includes all our possible types, and set it as a default value for `type` argument, or use your approach. I am up for going with your approach since I think it is reasonable to pass `type=str` to `EnvWithSibilings`. Any config going from an env var is assumed to be `str` type. Since `type` is a required argument, we have to explicitly pass it. So I don't even think we need to put any comment here. Or, at least, we could correct it a bit not saying that "we have to pass anything here...".

Reply:
Rephrased the in-code comment:
```
    # 'type' is a mandatory parameter for '__init_subclasses__', so we have to pass something here,
    # this doesn't force child classes to have 'str' type though, they actually can be any type
```